api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -252,12 +253,21 @@ private Runtime getOverrideFromListRuntimes(String googleProject) {
       runtime = new Runtime();
     }
 
-    if (runtime.getGceConfig() == null && runtime.getDataprocConfig() == null) {
-      throw new BadRequestException("Either a GceConfig or DataprocConfig must be provided");
+    int nullCount =
+        Booleans.countTrue(
+            runtime.getGceConfig() == null,
+            runtime.getDataprocConfig() == null,
+            runtime.getGceWithPdConfig() == null);
+    if (nullCount == 3) {
```

Comment:
Optional: I would be happy to converge these into a single exception case: "Exactly one of ..."

Also, another way of potentially writing this:


```
int configCount = Stream.of(
    runtime.getGceConfig(), 
    runtime.getDataprocConfig(),
    runtime.getGceWithPdConfig())
  .filter(c -> c != null)
  .count();
```